### PR TITLE
refactor(locale)!: use singular locale definition keys

### LINKED ISSUE #2911: Use singular locale definition keys
Followup for #2910 

- #2910 

Currently we have some locale files that use plural naming, and some that use singular naming.

| singular | plural |
| ------------------------------------------- | -------------------------------------------- |
| `faker.definitions.science.chemical_element` | `faker.definitions.system.directory_paths`   |

We should decide on one naming scheme and stick to that.

Currently, most definitions are singular.

### PR DESCRIPTION
**Pull request by SaurabhXRT**
### Overview

This PR addresses issue #2911 by standardizing the locale definition keys to use singular naming. This change improves consistency across the codebase and aligns with the team decision discussed in the issue.

### Changes Made

- Updated `LoremDefinition` to use the singular key `word` instead of `words`.
- Modified the automatically generated `lorem` definition file to use `word`:
  - Changed the key from `words` to `word`.
  - Updated the import statement to match the current file structure.
- Updated `SystemDefinition` to use singular keys:
  - Changed `directory_paths` to `directory_path`.
  - Changed `mime_types` to `mime_type`.
  - Updated the import statements to match the current file structure.

### Testing

- Ran all existing tests to ensure no breakage.
- Updated tests where necessary to match the new naming convention.

### Additional Notes

Please review the changes and let me know if any further updates are needed. This change includes a breaking change label as it modifies the locale definition keys, which might affect backward compatibility.

### CODE REVIEW

File: src/locales/ar/lorem/index.ts [hunk 1]
```diff
@@ -3,10 +3,10 @@
  * Run 'pnpm run generate:locales' to update.
  */
 import type { LoremDefinition } from '../../..';
-import words from './words';
+import word from './words';
```
Comment (ST-DDT) [inline src/locales/ar/lorem/index.ts]: You have to rename the files as well, because these files are auto generated.

**Review by ST-DDT — CHANGES_REQUESTED** [review]

**Comment by ST-DDT** [comment]
Also please edit this section and add the renames:

https://github.com/faker-js/faker/blob/next/docs/guide/upgrading.md#renamed-locale-definitions

File: src/locales/ar/lorem/index.ts (hunk 1 above)
Comment (SaurabhXRT) [reply src/locales/ar/lorem/index.ts]: Thank you for pointing that out. I will proceed to rename the files to align with the singular naming convention.

Comment (SaurabhXRT) [reply src/locales/ar/lorem/index.ts]: I have renamed the file also and upgraded the upgrading.md file.

**Comment by ST-DDT** [comment]
Please make sure to run `pnpm run preflight`

File: docs/guide/upgrading.md [hunk 2]
```diff
@@ -482,13 +482,14 @@ Renamed deprecated locale aliases `cz`, `en_IND`, `ge` and removed `global`.
 
 ### Renamed Locale Definitions
 
-The following locale definitions have been renamed
+The following locale definitions have been renamed to use singular keys and file names:
```
Comment (xDivisionByZerox) [inline docs/guide/upgrading.md]: The information that the file names have changed should not be relevant to the end user. I think a more generic description would be fitting here.

```suggestion
The following locale definitions have been adjusted to align with fakers locale definition naming standard:
```

The title of this section should propably also be changed too 🤔

**Review by xDivisionByZerox — CHANGES_REQUESTED** [review]

File: docs/guide/upgrading.md (hunk 2 above)
Comment (SaurabhXRT) [reply docs/guide/upgrading.md]: I have run but I am getting this error " ERR_PNPM_READ_FROM_STORE  Unexpected token '', ""... is not valid JSON".
I am figuring out the error after that I will update.

**Comment by SaurabhXRT** [comment]
hey @matthewmayer i have some doubt i am facing error how can i connect with you ?
can you please check test/modules/lorem.spec.ts file what is wrong?

File: test/modules/lorem.spec.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ describe('lorem', () => {
             expect(words).toHaveLength(num);
 
             for (const word of words) {
-              expect(faker.definitions.lorem.words).toContain(word);
+              expect(faker.definitions.lorem.word).toContain(word);
             }
           }
         );
```
Comment (ST-DDT) [inline test/modules/lorem.spec.ts:95]: Line 109: 

````diff
- expect(faker.definitions.lorem.words).toContain(word);
+ expect(faker.definitions.lorem.word).toContain(word);
````

File: docs/guide/upgrading.md [hunk 4]
```diff
@@ -482,13 +482,14 @@ Renamed deprecated locale aliases `cz`, `en_IND`, `ge` and removed `global`.
 
 ### Renamed Locale Definitions
 
-The following locale definitions have been renamed
+The following locale definitions have been adjusted to align with fakers locale definition naming standard:
 
-| old                                         | replacement                                  |
+| Old                                         | Replacement                                  |
```
Comment (ST-DDT) [inline docs/guide/upgrading.md]: The other tables use locercase headers as well.

```suggestion
| old                                         | replacement                                  |
```

File: pnpm-lock.yaml
Comment (ST-DDT) [inline pnpm-lock.yaml:1]: Please revert the change to this file.

**Comment by ST-DDT** [comment]
Two more small changes and then this is good to go.

File: docs/guide/upgrading.md [hunk 5]
```diff
@@ -482,13 +482,14 @@ Renamed deprecated locale aliases `cz`, `en_IND`, `ge` and removed `global`.
 
 ### Renamed Locale Definitions
 
-The following locale definitions have been renamed
+The following locale definitions have been adjusted to align with fakers locale definition naming standard:
```
Comment (matthewmayer) [inline docs/guide/upgrading.md]:
```suggestion
The following locale definitions have been adjusted to align with Faker's locale definition naming standard:
```

File: pnpm-lock.yaml
Comment (SaurabhXRT) [reply pnpm-lock.yaml:1]: I didnt change anything in the pnpm-lock.yml I accidentally deleted that file so i have to add that again.

**Review by matthewmayer — DISMISSED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by matthewmayer — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Comment by xDivisionByZerox** [comment]
@SaurabhXRT Thank you for your contribution ❤️

**Pull request merged**